src/interpreter/analytics.rs

```diff
@@ -0,0 +1,154 @@
+use std::collections::HashSet;
+
+use crate::bounding_box::BoundingBox;
+use crate::interpreter::{LogMessage, MeshArrayValue};
+use crate::mesh::{analysis, Mesh, UnorientedEdge};
+
+pub fn report_mesh_analysis(mesh: &Mesh) -> Vec<LogMessage> {
```

Comment:
This will work, but it would be slightly better if propagated the logger as a parameter instead. We wouldn't need to allocate the temporary vector in this case. I think the log function can possibly even have the same signature.

I know you asked me about this IRL and I said something else, but the difference here is that these 2 are ad-hoc functions that will probably always live in the application. I would still be against passing the logger to library functions.